docs/src/details/common-widget-features/cache_framework/implementation_details.rst

```diff
@@ -0,0 +1,262 @@
+       lv_rb_t rb;                /* Red-Black Tree */
+       lv_ll_t ll;                /* Doubly linked list */
+       lv_lru_rb_get_size_cb_t get_data_size_cb; /* Callback function to get data size */
+   } lv_lru_rb_t;
+
+.. mermaid::
+   :zoom:
+
+   %%{init: {'theme':'neutral'}}%%
+   graph TD
+       subgraph Red-Black Tree
+           RB["Red-Black Tree Root"] --> A["Node A"] & B["Node B"]
+           A --> C["Node C"] & D["Node D"]
+           B --> E["Node E"] & F["Node F"]
+       end
+
+       subgraph Doubly Linked List
+           LL["List Head"] --> B
+           B --> E
+           E --> A
+           A --> D
+           D --> C
+           C --> F["List Tail"]
+       end
+
+       A -.-> A_LL["List Node A"]
+       B -.-> B_LL["List Node B"]
+       C -.-> C_LL["List Node C"]
+       D -.-> D_LL["List Node D"]
+       E -.-> E_LL["List Node E"]
+       F -.-> F_LL["List Node F"]
+
+1.1.2 LRU Operation Flow
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. **Lookup Operation**:
+   - Use the red-black tree for fast cache entry lookup
+   - If found, move the entry to the head of the list (mark as recently used)
+   - Return the found entry
+
+2. **Add Operation**:
+   - Check if there is enough space
+   - If space is insufficient, evict the entry at the tail of the list (least recently used)
+   - Create a new entry and add it to the red-black tree and head of the list
+   - Return the newly added entry
+
+3. **Eviction Operation**:
+   - Get the least recently used entry from the tail of the list
+   - Check if the entry can be evicted (reference count is zero)
+   - If evictable, remove the entry from the red-black tree and linked list
+   - Mark the entry as invalid, but do not free memory (wait until reference count is zero to free)
+
+1.2 Count-based LRU Cache
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The count-based LRU cache (`lv_cache_class_lru_rb_count`) limits the number of entries in the cache. When the number of entries in the cache reaches the maximum value, it evicts the least recently used entries.
+
+.. code-block:: c
+
+   typedef struct _lv_lru_rb_count_t {
+       lv_lru_rb_t lru_rb;        /* LRU red-black tree base structure */
+   } lv_lru_rb_count_t;
+
+1.2.1 Space Check
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: c
+
+   static lv_cache_reserve_cond_t lru_rb_count_reserve_cond_cb(lv_cache_t * cache, const void * key, uint32_t reserved_size, void * user_data)
+   {
+       lv_lru_rb_count_t * count_cache = (lv_lru_rb_count_t *)cache;
+       
+       if(cache->size < cache->max_size) {
+           return LV_CACHE_RESERVE_COND_OK;
+       }
+       
+       return LV_CACHE_RESERVE_COND_NEED_VICTIM;
+   }
+
+1.3 Size-based LRU Cache
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The size-based LRU cache (`lv_cache_class_lru_rb_size`) limits the total memory size occupied by the cache. When the cache memory usage reaches the maximum value, it evicts the least recently used entries until there is enough space to accommodate new entries.
```

Comment:
I think these details should be documented in the code. Not by writing pages of comments but a text similar to this should be added before `lv_cache_class_lru_rb_size` in the code..